### DISCUSSION TITLE
Display search query in URL

### DISCUSSION THREAD

**Discussion by xplosionmind** [Pagefind/pagefind#629] [Q&A] [answered]
Hi all!

I would like to exploit Pagefind also from the search page URL.

An explanatory example: when I visit [the search page of Eleventy’s documentation](https://www.11ty.dev/docs/search/), while I type the search query the URL is automatically updated. If I search “if”, the page URL becomes: `https://www.11ty.dev/docs/search/?q=if`.

Is there a way I could make Pagefind on my website do the same?

**Comment by nhoizey** [comment]
You can use [`processTerm`](https://pagefind.app/docs/ui/#process-term):

```javascript
processTerm: function (term) {
  history.replaceState({}, "", `?q=${encodeURIComponent(term)}`);
  return term;
},
```

**Comment by nhoizey** [comment, marked as answer]
@xplosionmind if you add the query string to the URL, you should also use it to run the search when arriving on the page.

Here's how I did it on [my search page](https://nicolas-hoizey.photo/search/?q=%E2%B4%B0%E2%B4%B3%E2%B4%B0%E2%B4%B7%E2%B5%89%E2%B5%94):

```javascript
const params = new URLSearchParams(window.location.search);
if (params.has('q')) {
  search_ui.triggerSearch(params.get('q'));
}
```

My full code here: https://github.com/nhoizey/nicolas-hoizey.photo/blob/main/src/search/index.njk

**Reply by xplosionmind** [reply]
Thanks a lot, @nhoizey! Your search page code was super useful and insightful.

**Reply by xplosionmind** [reply]
@nhoizey do you have any idea how I can also include filters in the URL?

**Reply by nhoizey** [reply]
@xplosionmind you can't right now, but you can add your vote to this issue I opened a few weeks ago: https://github.com/CloudCannon/pagefind/issues/588
